teuthology/suite/__init__.py

```diff
@@ -148,11 +140,59 @@ def main(args):
                     conf.archive_upload_url)
 
 
-def get_rerun_filters(name, statuses):
+def get_rerun_conf_overrides(conf):
     reporter = ResultsReporter()
-    run = reporter.get_run(name)
+    run = reporter.get_run(conf.rerun)
+
+    conf.suite = normalize_suite_name(run['suite'])
+
+    try:
+        job0 = run['jobs'][0]
+    except IndexError:
+        job0 = None
+
+    seed = None if job0 is None else job0.get('seed')
+    if conf.seed >= 0 and conf.seed != seed:
+        log.error('--seed %s does not match with rerun seed: %s',
+                  conf.seed, seed)
+        sys.exit(1)
+    else:
+        log.info('Using rerun seed=%s', seed)
+        conf.seed = seed
+
+    subset = None if job0 is None else tuple(map(int, job0.get('subset').split('/')))
```

Comment:
i see `--rerun` failing here with `'NoneType' object has no attribute 'split'`. i generally run the rgw suite without `--subset`

opened https://tracker.ceph.com/issues/66302